src/sonic-yang-models/yang-models/sonic-bgp-common.yang

```diff
@@ -0,0 +1,540 @@
+            }
+            enum external {
+                description "External peer";
+            }
+        }
+    }
+
+    typedef bgp_tx_add_paths_type {
+        type enumeration {
+            enum tx_all_paths {
+                description
+                    "Send multiple path advertisements for an NLRI from 
+                    the neighbor or group";
+            }
+            enum tx_best_path_per_as {
+                description
+                    "Send only best path per AS advertisements for an NLRI from 
+                    the neighbor or group";
+            }
+        }
+        description
+            "Type to describe the add paths advertisement method.";
+    }
+
+    typedef bgp_community_type {
+        type enumeration {
+            enum standard {
+                description "Standard communities";
+            }
+
+            enum extended {
+                description "Extended communities";
+            }
+
+            enum both {
+                description "Both standard and extended communities";
+            }
+
+            enum large {
+                description "Large communities";
+            }
+
+            enum all {
+                description "Standard, Extended and Large communities";
+            }
+
+            enum none {
+                description "No community attribute";
+            }
+        }
+        description
+            "type describing variations of community attributes";
+    }
+
+    typedef sonic_bgp_direction {
+        type enumeration {
+            enum in {
+                description
+                    "Refers to information received from the BGP peer";
+            }
+            enum out {
+                description
+                    "Refers to information advertised to the BGP peer";
+            }
+        }
+        description
+            "Type to describe the direction";
+    }
+
+    typedef sonic_bgp_orf {
+        type enumeration {
+            enum send {
+                description
+                    "Capability to receive the outbound route filtering from this neighbor";
+            }
+            enum receive {
+                description
+                    "Capability to send the outbound route filtering to this neighbor";
+            }
+            enum both {
+                description
+                    "Capability to send and receive the outbound route filtering to/from this neighbor";
+            }
+        }
+        description
+            "Type to describe the BGP ORF(Outbound route filtering) capability";
+    }
+
+    grouping sonic-bgp-cmn {
+
+        leaf local_asn {
+            type uint32 {
+                range "1..4294967295";
+            }
+            description "Local AS number";
+        }
+
+        leaf name {
+            type string;
+            description "Peer description";
+        }
+
+        leaf asn {
+            type uint32 {
+                range "1..4294967295";
```

Comment:
should this be part of the BGP neighbor block instead ?

Reply:
This is common for both BGP neighbor and peer-group.